workflow/scripts/modify_district_heat_share.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+import logging
+
+logger = logging.getLogger(__name__)
+import geopandas as gpd
+import pandas as pd
+from shapely.geometry import Point
+
+
+def cluster_egon(heat_techs):
+    """
+    Map NUTS3 regions of egon data to corresponding clusters according to
+    maximum overlap.
+    """
+
+    regions_onshore = gpd.read_file(snakemake.input.regions_onshore)
+    regions_onshore.set_index("name", inplace=True)
+    # Map NUTS3 regions of egon data to corresponding clusters according to maximum overlap
+
+    heat_techs["cluster"] = heat_techs.apply(
+        lambda x: regions_onshore.geometry.intersection(x.geometry).area.idxmax(),
+        axis=1,
+    )
+
+    # Group and aggregate by cluster
+    heat_techs_clustered = heat_techs.groupby("cluster").sum(numeric_only=True)
+
+    return heat_techs_clustered
+
+
+def update_urban_loads_de(heat_techs_clustered):
+    """
+    Update district heating demands of clusters according to shares in egon
+    data on NUTS3 level for Germany.
+
+    Other heat loads are adjusted accodingly to ensure consistency of
+    the nodal heat demand.
+    """
+
+    nodal_dh_shares = heat_techs_clustered[
```

Comment:
did you check the overall German DH share? Does this fit to other data for German district heating share?

Reply:
Combining statistics from AGEB/UBA and BDEW, the district heating demand share in 2021 was 14% (126 TWh). (Sources: [BDEW](https://www.bdew.de/presse/presseinformationen/zdw-fernwaerme-126-milliarden-kilowattstunden/) & [UBA](https://www.umweltbundesamt.de/daten/energie/energieverbrauch-fuer-fossile-erneuerbare-waerme)).

According to AGFW, 14% of German households are connected to district heating grids. (Source: [AGFW](https://www.agfw.de/energiewirtschaft-recht-politik/energiewende-politik/system-kwk-fernwaerme)).

After adjusting the district heating demand shares, the model results show an exact match of 14% for both the demand and the percentage of households connected to district heating grids in 2020. Prior to the adjustment, the share was 9%.

Reply:
Great! Can you double check, that the 14% refer to residential and service sector?